File: Linguagem de Programação I/Extras/Beecrowd/1124.py
import math

while True:
    L, C, R1, R2 = map(int, input().split())
    if L == C == R1 == R2 == 0:
        break

    if max(R1, R2) * 2 > min(L, C):
        print('N')
        continue

    if (R1 + R2) ** 2 <= (L - R1 - R2) ** 2 + (C - R1 - R2) ** 2:
        print('S')
    else:
        print('N')

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1124.php
<?php
while (true) {
    $input = fgets(STDIN);
    if ($input === false) {
        break;
    }

    $inputs = explode(' ', trim($input));
    $L = intval($inputs[0]);
    $C = intval($inputs[1]);
    $R1 = intval($inputs[2]);
    $R2 = intval($inputs[3]);

    if ($L == 0 && $C == 0 && $R1 == 0 && $R2 == 0) {
        break;
    }

    if (max($R1, $R2) * 2 > min($L, $C)) {
        echo "N\n";
        continue;
    }

    if (pow($R1 + $R2, 2) <= pow($L - $R1 - $R2, 2) + pow($C - $R1 - $R2, 2)) {
        echo "S\n";
    } else {
        echo "N\n";
    }
}
?>